### PR TITLE
Ruby: ignore meta ci messages in testing

### LINKED ISSUE #41: Ruby: Disregard meta messages when running locally/remotely
### 👓 What did you see?

In Ruby, we disregard **additional** CI messages. But we don't disregard **missing** CI messages.

Implement the same logic again

### ✅ What did you expect to see?

```
missing_keys = (expected_keys - found_keys).reject { |key|
  ENV['CI'] && found.class == Cucumber::Messages::Meta && key == :ci
}

extra_keys = (found_keys - expected_keys).reject { |key|
  ENV['CI'] && found.class == Cucumber::Messages::Meta && key == :ci
}
```

### 🔬 How could we reproduce it?

1. Install '...' version '...'
2. Create a file called '....'
3. Run command '....'
4. See error '....'

### 📚 Any additional context?

----
 
*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/bug_report.md)*

### PR DESCRIPTION
**Pull request by luke-hill**
### 🤔 What's changed?

Add a test and new functionality to ignore **both** extra and missing keys that are of the Meta Envelope

### ⚡️ What's your motivation? 

Fixes #41 

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)
- 
### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by luke-hill** [comment]
Is this good to merge also (I've put changelogs in a bit prematurely for both fixes).

File: ruby/lib/keys_checker.rb [hunk 1]
```diff
@@ -12,11 +12,13 @@ def compare(found, expected)
 
       return errors if found_keys.sort == expected_keys.sort
 
-      missing_keys = (expected_keys - found_keys)
+      missing_keys = (expected_keys - found_keys).reject do |key|
+        ENV['CI'] && found.class == Cucumber::Messages::Meta && key == :ci
```
Comment (davidjgoss) [inline ruby/lib/keys_checker.rb:16]: Do we want to just do this regardless of the `CI` env var? I recall your machine happened to have something set for this, but what if it doesn't?

Comment (luke-hill) [reply ruby/lib/keys_checker.rb:16]: In hindsight yes, that's not a negative. But what it means is running **on** ci won't check this key.

But if this key is just jargon from the CI executor then it seems reasonable to just never match on it. However then we enter the realm of not testing a part of the schema ever? Thoughts?

Comment (davidjgoss) [reply ruby/lib/keys_checker.rb:16]: I would say never match on it, JS and JVM do basically the same and the `ci-environment` library does thorough enough testing of its own. Really the focal point of CCK is the messages about the tests themselves.

**Pull request merged**